### PR TITLE
[decimal] rounding bug Round(1.005, 2) = 1.01

### PR DESCRIPTION
**Pull request by haf**
Here you go! A failing test!

### CODE REVIEW

**Comment by monoadmin** [comment]
Hello! I'm the build bot for the Mono project. I need approval from a Mono team member to build this pull request. A team member should reply with "approve" to approve a build of this pull request, "whitelist" to whitelist this and all future pull requests from this contributor, or "build" to explicitly request a build, even if one has already been done. Contributors can ignore this message.

File: mcs/class/corlib/Test/System/DecimalTest.cs [hunk 1]
```diff
@@ -882,6 +882,8 @@ public void TestRound ()
 				{1.234567890m, 2, 1.23m}, {1.23450000001m, 3, 1.235m}, 
 				{1.2355m, 3, 1.236m}, 
 				{1.234567890m, 4, 1.2346m}, {1.23567890m, 2, 1.24m}, 
+				{1.00567890m, 2, 1.01m},
```
Comment (haf) [inline mcs/class/corlib/Test/System/DecimalTest.cs]: This line works

File: mcs/class/corlib/Test/System/DecimalTest.cs [hunk 2]
```diff
@@ -882,6 +882,8 @@ public void TestRound ()
...
+				{1.005m, 2, 1.01m},
```
Comment (haf) [inline mcs/class/corlib/Test/System/DecimalTest.cs]: This doesn't.

Comment (akoeplinger) [inline mcs/class/corlib/Test/System/DecimalTest.cs]: @haf I just tried on MS.NET and it fails there as well.

Comment (haf) [inline mcs/class/corlib/Test/System/DecimalTest.cs]: It's a bug though, right?

This is how ruby does it (round takes # digits (not decimals)):

``` ruby
0.005.round(2)
# => 0.01
```

Clojure:

``` clojure
(format "%.2f" 1.005)
; => "1.01"
```

Comment (haf) [inline mcs/class/corlib/Test/System/DecimalTest.cs]: StringFormat on Mono:

``` fsharp
> String.Format("{0:F2}", 1.005);;
val it : string = "1.01"
```

Comment (haf) [inline mcs/class/corlib/Test/System/DecimalTest.cs]: Here's a funky one, again mono, but with what should be the 'default' argument:

``` fsharp
Math.Round(1.005M, 2, MidpointRounding.AwayFromZero);;
val it : decimal = 1.01M
```

**Comment by haf** [comment]
Ruby is also wrong btw, if using `1.005`:

``` ruby
1.005.round(2)
# => 1.0
```

File: mcs/class/corlib/Test/System/DecimalTest.cs (hunk 2 above)
Comment (akoeplinger) [inline mcs/class/corlib/Test/System/DecimalTest.cs]: @haf ToEven **is** the default according to the [docs](https://msdn.microsoft.com/en-us/library/system.math.round(v=vs.110).aspx):

> MidpointRounding  
 
> A flag that specifies the convention to use for rounding midpoint values. See Midpoint values and rounding conventions.

> Default value: MidpointRounding.ToEven

Comment (haf) [inline mcs/class/corlib/Test/System/DecimalTest.cs]: That's true. Confusing that parts of the framework (round) uses ToEven, but String.Format uses AwayFromZero!

And ruby does this:

```
irb(main):001:0> 1.005.round(2)
=> 1.0
irb(main):002:0> 0.005.round(2)
=> 0.01
```

Is that correct then? In that case, it seems like ruby uses both modes?

Closing :)

Comment (akoeplinger) [inline mcs/class/corlib/Test/System/DecimalTest.cs]: Yeah, I agree this is confusing.

Let me point you to this excellent [SO answer](http://stackoverflow.com/questions/2226081/why-does-net-use-a-rounding-algorithm-in-string-format-that-is-inconsistent-wit) by Eric Lippert:
> Perhaps this is yet another case of 25 years of backwards compatibility.

:smile:

**Pull request closed without merging**